starport/cmd/network_chain_init.go

```diff
@@ -0,0 +1,193 @@
+		return err
+	}
+	if !getYes(cmd) && exist {
+		prompt := promptui.Prompt{
+			Label: fmt.Sprintf("The chain has already been initialized under: %s. Would you like to overwrite the home directory",
+				chainHome,
+			),
+			IsConfirm: true,
+		}
+		s.Stop()
+		if _, err := prompt.Run(); err != nil {
+			fmt.Println("said no")
+			return nil
+		}
+		s.Start()
+	}
+
+	nb, err := newNetwork(cmd, network.CollectEvents(ev))
+	if err != nil {
+		return err
+	}
+
+	// initialize the blockchain from the launch ID
+	initOptions := initOptionWithHomeFlag(cmd, []network.InitOption{network.MustNotInitializedBefore()})
+	sourceOption := network.SourceLaunchID(launchID)
+	blockchain, err := nb.Blockchain(cmd.Context(), sourceOption, initOptions...)
+	if err != nil {
+		return err
+	}
+
+	if err := blockchain.Init(cmd.Context()); err != nil {
+		return err
+	}
+
+	// ask validator information
+	v, err := askValidatorInfo()
+	if err != nil {
+		return err
+	}
+
+	// ask user for mnemonic if not provided and recover option defined
+	if mnemonic == "" && recover {
+		cliquiz.Ask(cliquiz.NewQuestion("Enter your account mnemonic",
+			&mnemonic,
+			cliquiz.DefaultAnswer(""),
+			cliquiz.Required(),
+		))
+	}
+
+	acc, gentxPath, err := blockchain.InitAccount(cmd.Context(), v, keyName, mnemonic)
+	if err != nil {
+		return err
+	}
+	fmt.Printf(`Account created
+Address: %s
+Mnemonic: %s
+`, acc.Address, acc.Mnemonic)
+	fmt.Printf("Gentx generated: %s\n", gentxPath)
+
+	return nil
+}
+
+// askValidatorInfo prompts to the user questions to query validator information
+func askValidatorInfo() (v chain.Validator, err error) {
+	questions := append([]cliquiz.Question{},
+		cliquiz.NewQuestion("Staking amount",
+			&v.StakingAmount,
+			cliquiz.DefaultAnswer("95000000stake"),
+			cliquiz.Required(),
+		),
+		cliquiz.NewQuestion("Moniker",
+			&v.Moniker,
+			cliquiz.DefaultAnswer("mynode"),
+			cliquiz.Required(),
+		),
+		cliquiz.NewQuestion("Commission rate",
+			&v.CommissionRate,
+			cliquiz.DefaultAnswer("0.10"),
+			cliquiz.Required(),
+		),
+		cliquiz.NewQuestion("Commission max rate",
+			&v.CommissionMaxRate,
+			cliquiz.DefaultAnswer("0.20"),
+			cliquiz.Required(),
+		),
+		cliquiz.NewQuestion("Commission max change rate",
+			&v.CommissionMaxChangeRate,
+			cliquiz.DefaultAnswer("0.01"),
+			cliquiz.Required(),
+		),
+		cliquiz.NewQuestion("Min self delegation",
+			&v.MinSelfDelegation,
+			cliquiz.DefaultAnswer("1"),
+			cliquiz.Required(),
+		),
+		cliquiz.NewQuestion("Gas prices",
+			&v.GasPrices,
+			cliquiz.DefaultAnswer("0.025stake"),
+			cliquiz.Required(),
+		),
+		cliquiz.NewQuestion("Details", &v.Details),
+		cliquiz.NewQuestion("Identity", &v.Identity),
+		cliquiz.NewQuestion("Website", &v.Website),
+	)
+	return v, cliquiz.Ask(questions...)
+}
+
+// checkChainHomeExist checks if a home with the provided launchID already exist
+func checkChainHomeExist(launchID uint64) (string, bool, error) {
+	home, err := network.ChainHome(launchID)
```

Comment:
Can we refactor and use blockchain.IsHomeDirExist instead?

Reply:
You must currently fetch the launch ID then fetch the source to initialize `blockchain`. This one allows doing the check directly
We can implement a method in `network` used by both parts

Another solution would be to refactor `Blockchain` to allow initializing a blockchain without source but I'm afraid we would make the package overengineered

Reply:
Can we implement a network.IsChainHomeExists then?